docs/DeveloperGuide.md

```diff
@@ -1,34 +1,511 @@
+The UI class will then parse the user input into the ControlManager class where the command will be extracted and processed by the CommandParser class.
+According to which category they belong to i.e class,cca,test,tuition etc, they are sent to their respective category managers.
+In each manager, the user input is processed and made into a new Event item by the Event class.
+The result is subsequently outputted by the UI class to the user. 
+
+Upon completion of this feature, it returns a boolean value “true” to the active flag in UserInterface.java to allow the continuous usage of the program.
+
+
+### Delete feature
+
+![deletecomponent](https://raw.githubusercontent.com/AY2021S1-CS2113T-W12-4/tp/master/docs/diagram/DeleteFeatureDiagram.png)
+
+Figure 14. Sequence Diagram of the delete feature
+
+Firstly, when the user enters delete <event> <number>, the input will be read in by the UI class.
+The UI class will then parse the user input into the ControlManager class where the command will be extracted and processed by the CommandParser class. The result is passed into ModelExtractor Class.
+According to which category they belong to i.e. class, cca, test, tuition etc, they are sent to their respective category managers. In each manager, the user input is processed and the Event item would be deleted by the Event class.
+The result is subsequently outputted by the UI class to the user. 
+
+Upon completion of this feature, it returns a boolean value “true” to the active flag in UserInterface.java to allow the continuous usage of the program. 
+
+### List feature
+
+#### List contact / quiz
+
+![listcontact](https://raw.githubusercontent.com/AY2021S1-CS2113T-W12-4/tp/master/docs/diagram/ListContact.png)
+
+Figure 15. Sequence diagram for listing quizzes
+
+The list command will invoke the `ContactManager` or `QuizManager` class’s list() method respectively.
+If the ArrayList is empty, it will inform the user that there is no data to display.
+The method will then loop through the ArrayList and convert it into its string representation.
+This is then passed to the `UserInterface#printArray()` to be printed out to the user.
+
+#### List event (<date> / today / week)
+While the back-end data processing is the same for all three types of list requests, the list event week request requires a different front-end class to display to the user.
+As such, we divide this section into 2 sub-sections, with List event (<date>/today) in the first subsection, and List event week in the next subsection.
+
+The execution of the ListCommand will cause the `EventManager` class to invoke `EventManager#listSchedule()`.
+This will then create an instance of the `ListSchedule` class, with all the classes, ccas, tests and tuitions data
+as its attributes.
+
+#### List event (<date> / today)
+
+![listevent](https://raw.githubusercontent.com/AY2021S1-CS2113T-W12-4/tp/master/docs/diagram/ListEvent.png)
```

Comment:
Maybe there can be a return arrow here.
![image](https://user-images.githubusercontent.com/26886705/97534658-55acc500-19f5-11eb-9cd9-df5a4fb79561.png)